BranchMetrics/ios-branch-deep-linking-attribution
# [SDK-2458] Implement Consumer Protection Preferences

## Reference
SDK-2458 -- [iOS] Implement Consumer Protection Preferences

## Summary
Added a new method for setting the consumer protection preference, `setConsumerProtectionPreference`. This value can be set and changed at any time, but persists across sessions. This change also deprecated the `setTrackingDisabled()` method since  setting the preference to `BranchConsumerProtectionPreferenceTrackingDisabled` will perform the same logic.

## Motivation

To give clients more control over what end user information is tracked or used for attribution.

## Type Of Change


## Testing Instructions

Call `[[Branch getInstance] setConsumerProtectionPreference:BranchConsumerProtectionPreferenceAnalyticsOnly];`, or any other preference, at any point in the app and then observe the respective field appearing in the following requests, like `"protection_preference" : 2,`









cc @BranchMetrics/saas-sdk-devs for visibility.
